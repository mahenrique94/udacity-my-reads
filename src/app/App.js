import React, { Component } from "react"

class App extends Component {
    render() {
        return <h1>My Reads</h1>
    }
}

export default App
