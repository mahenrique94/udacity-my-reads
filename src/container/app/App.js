import React, { Fragment } from "react"
import PropTypes from "prop-types"

import { Container, Section } from "react-bulma-components"

const App = ({ children }) => (
    <Fragment>
        <Section>
            <Container>
                { children }
            </Container>
        </Section>
    </Fragment>
)

App.propTypes = {
    children: PropTypes.object.isRequired
}

export default App
