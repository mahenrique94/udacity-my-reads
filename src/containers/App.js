import PropTypes from "prop-types"
import React, { Component, Fragment } from "react"

import { getAll } from "@/books/api"

import BooksContext from "@/books/contexts/BooksContext"

import { loading } from "hoc/loading"

class App extends Component {

    static propTypes = {
        children: PropTypes.oneOfType([
            PropTypes.array,
            PropTypes.object
        ]).isRequired,
        shelfs: PropTypes.object.isRequired
    }

    constructor(props) {
        super(props)
        this.state = {
            shelfs: props.shelfs,
            updateShelfs: this.updateShelfs
        }
    }

    render() {
        const { children } = this.props
        return (
            <Fragment>
                <BooksContext.Provider value={ this.state }>
                    { children }
                </BooksContext.Provider>
            </Fragment>
        )
    }

    updateShelfs = newShelfs => this.setState({ shelfs: newShelfs })

}

export default loading(getAll)(App)
