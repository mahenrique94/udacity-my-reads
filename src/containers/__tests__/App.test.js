import React from "react"
import renderer from "react-test-renderer"

import App from "../App"

describe("Tests for App component", () => {

    it.skip("Should rendering without crash", () => {
        const tree = renderer.create(
            <App>
                <h1>Should rendering without crash</h1>
            </App>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

})
