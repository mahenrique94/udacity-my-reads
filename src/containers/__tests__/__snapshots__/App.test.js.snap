// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for App component Should rendering without crash 1`] = `
<div
  className="loader sc-bdVaJa goZOWA"
  style={Object {}}
/>
`;
